node/test/unit/tessel.js

```diff
@@ -1616,3 +1625,373 @@ exports['Tessel.SPI'] = {
     test.done();
   }
 };
+
+exports['Tessel.Wifi'] = {
+  setUp: function(done) {
+    this.Port = sandbox.stub(Tessel, 'Port');
+    this.fsWrite = sandbox.stub(fs, 'writeFile');
+    this.exec = sandbox.stub(childProcess, 'exec', (cmd, callback) => {
+      callback();
+    });
+    this.tessel = new Tessel();
+    done();
+  },
+
+  tearDown: function(done) {
+    Tessel.instance = null;
+    sandbox.restore();
+    done();
+  },
+
+  initialized: function(test) {
+    test.expect(3);
+
+    test.equal(this.tessel.network.wifi.isBusy, false, 'not busy by default');
+    test.equal(this.tessel.network.wifi.isConnected, false, 'not connected by default');
+    test.deepEqual(this.tessel.network.wifi.settings, {}, 'no setings by default');
+
+    test.done();
+  },
+
+  connect: function(test) {
+    test.expect(6);
+
+    var settings = {
+      ssid: 'TestNetwork',
+      password: 'TestPassword',
+      security: 'psk2'
+    };
+    var ip = '192.168.1.101';
+    var network = {
+      ssid: 'TestNetwork',
+      strength: '30/80'
+    };
+
+    this.exec.restore();
+    this.exec = sandbox.stub(childProcess, 'exec', (cmd, callback) => {
+      if (cmd === 'ifconfig wlan0') {
+        callback(null, ip);
+      } else if (cmd === 'ubus call iwinfo info {"device":"wlan0"}') {
+        callback(null, JSON.stringify(network));
+      } else {
+        callback();
+      }
+    });
```

Comment:
This a small refactoring suggestion: you seem to redefine `this.exec` with each function but with essentially the same body contents. You might want to bind the variables you return to `this` and just modify those rather than copying the same function.

```.js
setUp: function(done) {
...
this.exec = sandbox.stub(childProcess, 'exec', (cmd, callback) => {
       if (cmd === 'ifconfig wlan0') {
         callback(null, this.ip);
       } else if (cmd === 'ubus call iwinfo info {"device":"wlan0"}') {
         callback(null, JSON.stringify(this.network));
       } else {
         callback();
       }
     });
...
});

exports['some test'] = function(test) {
  this.ip = '192.168.1.101';
  this.network = { ssid:'testNetwork', strength: '30/80'};
  ...
};